automotive/src/main/java/com/ixam97/carStatsViewer/dataManager/DataCollector.kt

```diff
@@ -224,17 +222,14 @@ class DataCollector : Service() {
         saveTripDataTimerHandler.postDelayed(saveTripDataTask, AUTO_SAVE_INTERVAL_MILLIS)
 
         liveDataTimerHandler = Handler(Looper.getMainLooper())
-        abrpLiveDataTask = CarStatsViewer.liveDataApis[0].createLiveDataTask(
-            DataManagers.CURRENT_TRIP.dataManager,
-            liveDataTimerHandler,
-            LIVE_DATA_TASK_INTERVAL
-        )!!
-        httpLiveDataTask = CarStatsViewer.liveDataApis[1].createLiveDataTask(
-            DataManagers.CURRENT_TRIP.dataManager,
-            liveDataTimerHandler,
-            LIVE_DATA_TASK_INTERVAL
-        )!!
-        liveDataTimerHandler.post(httpLiveDataTask)
+        for (liveDataAPI in CarStatsViewer.liveDataApis) {
+            val task = liveDataAPI.createLiveDataTask(
+                DataManagers.CURRENT_TRIP.dataManager,
+                liveDataTimerHandler,
+                LIVE_DATA_TASK_INTERVAL
+            )!!
```

Comment:
Some APIs may not use a time interval to send data, but use other triggers. That's why `createLiveDataTask` can return `null`. So this should be handled here.

Reply:
See df427edcdea223f35ee5c74b4e57f1b8f6cd2975